packages/firebase_auth/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java

```diff
@@ -381,168 +345,70 @@ private void handleDelete(MethodCall call, Result result, FirebaseAuth firebaseA
         .addOnCompleteListener(new TaskVoidCompleteListener(result));
   }
 
-  private void handleSignInWithEmailAndPassword(
-      MethodCall call, Result result, FirebaseAuth firebaseAuth) {
-    Map<String, String> arguments = call.arguments();
-    String email = arguments.get("email");
-    String password = arguments.get("password");
-
-    firebaseAuth
-        .signInWithEmailAndPassword(email, password)
-        .addOnCompleteListener(new SignInCompleteListener(result));
+  private AuthCredential getCredential(Map<String, Object> arguments) {
+    AuthCredential credential;
+    Map<String, String> data = (Map<String, String>) arguments.get("data");
+    switch ((String) arguments.get("provider")) {
+      case "password":
```

Comment:
Could we use the constants here instead like we are doing for iOS?

`EmailAuthProvider.PROVIDER_ID`